ironfish/src/syncer.ts

```diff
@@ -341,21 +341,36 @@ export class Syncer {
     }
   }
 
-  async syncBlocks(peer: Peer, head: Buffer, sequence: number): Promise<void> {
-    this.abort(peer)
-
-    let count = 0
-    let skipped = 0
-
+  private async getBlocks(
+    peer: Peer,
+    start: Buffer,
+    limit: number,
+  ): Promise<{ blocks: Block[]; time: number }> {
     this.logger.info(
-      `Requesting ${this.blocksPerMessage} blocks starting at ${HashUtils.renderHash(
-        head,
-      )} (${sequence}) from ${peer.displayName}`,
+      `Requesting ${limit - 1} blocks starting at ${HashUtils.renderHash(start)} from ${
+        peer.displayName
+      }`,
     )
 
-    let blocksPromise = this.peerNetwork.getBlocks(peer, head, this.blocksPerMessage + 1)
+    return this.peerNetwork.getBlocks(peer, start, limit).catch((e) => {
```

Comment:
Should this `.catch` do the same thing as the `.catch` in `syncFrom`? I think it calls `close` on the Peer

Reply:
Yea usually this happens because the peer was closed but it could happen because of a timeout or another random error. I can add `peer.close` here too